payload/Library/Scripts/Install or Defer.sh

```diff
@@ -356,29 +355,39 @@ else
 fi
 
 # We need to be connected to the internet in order to download updates.
-if ping -q -c 1 208.67.222.222; then
+if /sbin/ping -q -c 1 208.67.222.222; then
     # Check if a custom CatalogURL is set and if it is available
-    SU_CATALOG=$(python -c 'from Foundation import CFPreferencesCopyAppValue; print CFPreferencesCopyAppValue("CatalogURL", "com.apple.SoftwareUpdate")')
-    if [[ "$SU_CATALOG" != "None" ]]; then
-        if /usr/bin/curl --user-agent "Darwin/$(uname -r)" -s --head "$SU_CATALOG" | grep "200 OK" > /dev/null; then
-            echo "[ERROR] Software update catalog can not be reached."
-            BAILOUT=true
+    # (deprecated in macOS 11+).
+    if [[ "$OS_MAJOR" -eq 10 && "$OS_MINOR" -lt 16 ]]; then
+        SU_CATALOG=$(/usr/bin/defaults read "/Library/Managed Preferences/com.apple.SoftwareUpdate" CatalogURL 2>"/dev/null")
+        if [[ "$SU_CATALOG" != "None" ]]; then
+            if /usr/bin/curl --user-agent "Darwin/$(/usr/bin/uname -r)" -s --head "$SU_CATALOG" | /usr/bin/grep "200 OK" >"/dev/null"; then
+                echo "❌ ERROR: Software update catalog can not be reached."
+                BAILOUT=true
+            fi
         fi
     fi
 else
-    echo "[ERROR] No connection to the Internet."
+    echo "❌ ERROR: No connection to the Internet."
     BAILOUT=true
 fi
 
 # If FileVault encryption or decryption is in progress, installing updates that
 # require a restart can cause problems.
-if fdesetup status | grep -q "in progress"; then
-    echo "[ERROR] FileVault encryption or decryption is in progress."
+if /usr/bin/fdesetup status | /usr/bin/grep -q "in progress"; then
+    echo "❌ ERROR: FileVault encryption or decryption is in progress."
     BAILOUT=true
 fi
 
 # If any of the errors above are present, bail out of the script now.
 if [[ "$BAILOUT" = "true" ]]; then
+    # Checks for StartInterval definition in LaunchDaemon.
+    START_INTERVAL=$(/usr/bin/defaults read "/Library/LaunchDaemons/$BUNDLE_ID.plist" StartInterval 2>"/dev/null")
+    if [[ -n "$START_INTERVAL" ]]; then
+        echo "Stopping due to errors, but will try again in $(convert_seconds "$START_INTERVAL")."
+    else
+        echo "Stopping due to errors."
+    fi
     echo "Stopping due to errors."
```

Comment:
This line looks redundant with the above.

Reply:
This is a restoration of previously removed code, but I suspect we could clean it up a bit. I'll take a look.